test/controllers/posts/help_test.rb

```diff
@@ -25,20 +25,20 @@ class PostsControllerTest < ActionController::TestCase
   test 'moderator help requires authentication' do
     get :document, params: { slug: posts(:mod_help_article).doc_slug }
     assert_response 404
-    assert_not_nil assigns(:post)
```

Comment:
I'm curious about the flip from not nil to nil in these tests without the names of the tests changing.  Were they backwards before?

Reply:
We've already [discussed this](https://discord.com/channels/634104110131445811/1306699834262032476/1317861103434469407) briefly with @ArtOfCode- in chat. Short version: the access check _used to_ happen _after_ the variable `@post` has been assigned, now it doesn't even happen if the user doesn't have access (as the check itself has been abstracted away), hence if the route responds with `404`, it is guaranteed to not assign the variable as well.